src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -338,18 +368,25 @@ static term setup_gpio_pin(Context *ctx, term gpio_pin_tuple, term mode_term)
         mode_atom = mode_term;
         if (UNLIKELY(!term_is_atom(mode_atom))) {
             AVM_LOGE(TAG, "GPIO Mode must be an atom ('input', 'output', 'output_od').");
-            return create_pair(ctx, ERROR_ATOM, INVALID_MODE_ATOM);
+            return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_mode_atom));
         }
         gpio_mode = ((uint8_t) interop_atom_term_select_int(pin_mode_table, mode_atom, ctx->global));
         if (UNLIKELY(gpio_mode == GPIO_INVALID_MODE)) {
             char *mode_string = interop_atom_to_string(ctx, mode_atom);
             AVM_LOGE(TAG, "Invalid gpio mode: %s", mode_string);
             free(mode_string);
-            return create_pair(ctx, ERROR_ATOM, INVALID_MODE_ATOM);
+            return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_mode_atom));
         }
         pull_up_down = GPIO_PUPD_NONE;
-        if (setup_output) {
```

Comment:
Was this one of the bugs?  Looks like it might have been initialized to false.

BTW, I understand the older C need to declare variable at the top of lexical bocks, but I prefer the new style of declaring them as close to usage as possible.  In this case, I realize it doesn’t make a difference, but there are other cases where it may.  Again, kind of a personal style thing, and up to you, but I think we use the newer convention throughout the codebase, and the spec and compilers have matured over the years, so no need to use to old Kernigan and Ritchie C :)

Reply:
Yes. This was not implied correctly the new driver was merged.

Thanks for the tip on the declaring variables... I actually went back and forth many times with where I placed the variables while working on the driver, and this set of corrections/cleanup. I will definitely keep that in mind in my future PRs.

Reply:
Since I needed to remove the unnecessary `struct ListHead *tmp` declaration already, I went ahead and delayed the declaration of variables until they are used.